CMakeLists.txt

```diff
@@ -24,6 +24,7 @@ mxnet_option(USE_OLDCMAKECUDA     "Build with old cmake cuda" OFF)
 mxnet_option(USE_NCCL             "Use NVidia NCCL with CUDA" OFF)
 mxnet_option(USE_OPENCV           "Build with OpenCV support" ON)
 mxnet_option(USE_OPENMP           "Build with Openmp support" ON)
+mxnet_option(USE_OPENMP_BUNDLED_LLVM "Build with bundled llvm openmp from 3rdparty" OFF)
```

Comment:
What happens when USE_OPENMP is OFF and USE_OPENMP_BUNDLED_LLVM ON?

Reply:
USE_OPENMP takes precedence, so no OMP is used.

Reply:
Is this documented somewhere? Can we set mxnet_option(USE_OPENMP   3RD_PARTY | BUILT_IN)?

I am not an expert in CMake so I can not really decide whether this is the most professional way. LGTM for now. But it's better to have someone from CI team to merge this PR.

Reply:
I used this for consistency. What you propose is also valid. I would do this incrementally as right now debugging is blocked.